oak-doc/src/site/markdown/security/authorization/bestpractices.md

```diff
@@ -0,0 +1,270 @@
+
+#### Built-in privileges
+
+JSR 382 defines a set of built-in privileges and how they apply to repository operations (see https://s.apache.org/jcr-2.0-javadoc/javax/jcr/security/Privilege.html).
+The default set has been extended by Oak to cover additional features outside of the scope defined by JCR (like e.g. index 
+or user management). The complete list can be found in [Privilege Management : The Default Implementation](../privilege/default.html). 
+
+The minimal set of privileges required for each repository operation can be looked up in [Mapping API Calls to Privileges](../privilege/mappingtoprivileges.html) 
+and [Mapping Privileges to Items](../privilege/mappingtoitems.html). 
+
+##### Privileges affecting the parent node
+
+Note in particular for add/removing a node `jcr:addChildNodes` and `jcr:removeChildNodes` are required on 
+the parent node respectively i.e. allowing for modification of the child-node collection. 
+In addition `jcr:removeNode` needs to be granted on the target node of the removal.
+
+Thus the following subtle difference apply when evaluation effective permissions vs. privileges (see 
+also [Permissions vs Privileges](../permission/permissionsandprivileges.html)) and exercises at 
+[L4_PrivilegesAndPermissionsTest.java](https://github.com/apache/jackrabbit-oak/blob/trunk/oak-exercise/src/test/java/org/apache/jackrabbit/oak/exercise/security/authorization/permission/L4_PrivilegesAndPermissionsTest.java)):
+
+        String parentPath = /content/parent
+        String toRemove = /content/parent/child
+        String toAdd = /content/parent/newchild
+        
+        Session session = ...
+                
+        # Testing Privileges
+        # ----------------------------------------------------------------------------------------
+        
+        AccessControlManager acMgr = session.getAccessControlManager();
+        Privilege jcrAddChildNodes = acMgr.privilegeFromName(Privilege.JCR_ADD_CHILD_NODES)
+        Privilege jcrRemoveChildNodes = acMgr.privilegeFromName(Privilege.JCR_REMOVE_CHILD_NODES)
+        Privilege jcrRemoveNode = acMgr.privilegeFromName(JCR_REMOVE_NODE)
+        
+        # test if (unspecified) child nodes can be added/removed from the parent
+        acMgr.hasPrivileges(parentPath, new Privilege[]{jcrAddChildNodes, jcrRemoveChildNodes}
+        
+        # test if existing child node can be removed
+        acMgr.hasPrivileges(toRemove, new Privilege[]{jcrRemoveNode}
+        
+        
+        # Testing Permissions (on the target node NOT on the parent)
+        # ----------------------------------------------------------------------------------------
+
+        # test if not-yet existing node could be added at /content/parent/newchild
+        session.hasPermission(toAdd, Session.ACTION_ADD_NODE)
```

Comment:
can you add an explicit ```assertTrue(...)``` or ```assertFalse(...)``` to make clear which result you expect? Same for the next statements.

Reply:
since the example doesn't come with any ac-setup i don't have an expectation on the result itself.... the goal was to illustrate the common confusion between hasPermission and hasPrivileges when it comes to adding/removing nodes.... but i will catch the return value in a hopefully meaningful variable.

Reply:
to be honest, I have not looked to much into this example. But maybe it might be best if we could expand this to a complete setup to illustrate the difference much better. Or even better, refer to an existing test.

Reply:
the exercise linked above will illustrate more or less the same.